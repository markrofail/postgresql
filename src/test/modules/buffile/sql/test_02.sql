BEGIN;
SELECT buffile_create();
SELECT buffile_seek(0, 8189);
-- Initialize the last 3 positions of the first buffer and the initial 3
-- positions of the 2nd buffer.
SELECT buffile_write('abcdef');
SELECT buffile_seek(0, 0);
-- Read the first buffer.
SELECT length(buffile_read(8192));
-- Only 3 bytes of the 2nd buffer should be fetched.
SELECT length(buffile_read(8192));
SELECT buffile_close();
COMMIT;
