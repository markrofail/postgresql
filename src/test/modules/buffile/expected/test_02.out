BEGIN;
SELECT buffile_create();
 buffile_create 
----------------
 
(1 row)

SELECT buffile_seek(0, 8189);
 buffile_seek 
--------------
            0
(1 row)

-- Initialize the last 3 positions of the first buffer and the initial 3
-- positions of the 2nd buffer.
SELECT buffile_write('abcdef');
 buffile_write 
---------------
             6
(1 row)

SELECT buffile_seek(0, 0);
 buffile_seek 
--------------
            0
(1 row)

-- Read the first buffer.
SELECT length(buffile_read(8192));
 length 
--------
   8192
(1 row)

-- Only 3 bytes of the 2nd buffer should be fetched.
SELECT length(buffile_read(8192));
 length 
--------
      3
(1 row)

SELECT buffile_close();
 buffile_close 
---------------
 
(1 row)

COMMIT;
